### ISSUE TITLE
Visualizations

### ISSUE THREAD

**Issue by MonZop** [MonZop/BioBlender#21]
Apparently the MainChain visualization does not work.
Main chain means only the N-CA-C atoms of every aminoacid. Sometimes it is good to see only these atoms, to get an idea of the general folding of the protein.

**Comment by MonZop** [comment]
Surface does work; however, if, after having calculated the surface, one want to see all atoms, or Main chain atoms only, the result is that Blender shows the surface again.

**Comment by zeffii** [comment]
when did this stop working?

**Comment by MonZop** [comment]
I noticed just now, a couple of days ago it was OK

**Comment by zeffii** [comment]
what about master?

**Comment by MonZop** [comment]
I can re-check, but it did work

**Comment by zeffii** [comment]
if master works, I can do a differential.

**Comment by MonZop** [comment]
Master _main_fixes dos not work

**Comment by zeffii** [comment]
and Master?

It would really help me if there were several smaller files that I could test, 02,03,04,05 just take forever to import.

**Comment by MonZop** [comment]
For this 06 should be sufficient: 

Master does not work :(

**Comment by zeffii** [comment]
current master is literally only three changes, all of which are necessary. When you close Blender and reopen can you take another note of the errors / screenshot please

**Comment by MonZop** [comment]
![master](https://cloud.githubusercontent.com/assets/10231334/6169305/0b917e7e-b2cd-11e4-8c23-773b3c7ef0a2.png)
![master1](https://cloud.githubusercontent.com/assets/10231334/6169307/1298e0ea-b2cd-11e4-894f-1758b7ff3ccb.png)

**Comment by zeffii** [comment]
untick those, press save User Settings and then press Ctrl+U  to store the change in the startup.blend

**Comment by MonZop** [comment]
![mainchain](https://cloud.githubusercontent.com/assets/10231334/6169460/e115ea76-b2cd-11e4-8fe1-0ede6e4814eb.png)
It works with Blender 2.70 and BioBlender_master

**Comment by zeffii** [comment]
I think I know why this is.. and it relates directly to the fix for the final conformation..  I will return to this later.

**Comment by zeffii** [comment]
the last lines of `02_4IHVonlyDNA.pdb` are 
```
ATOM   1088  C7   DT D  27      22.394  -3.354 124.626  1.00 35.02           C
TER    1088       DT D  27 
END
```
and `03_3IV5_DNA_proteinComplex`
```
HETATM 2620  O   HOH B 104      15.805   1.185  -2.720  1.00 24.86           O  
HETATM 2621  O   HOH B 105      35.829  17.152 -10.049  1.00 39.75           O  
HETATM 2622  O   HOH C  28      16.856  -0.777  37.497  1.00 44.91           O  
MASTER      412    0    0    9    4    0    0    6 2618    4    0   22          
END                                                                             
```
this is the nuance in pdb files that depend on the content.. and perhaps the program that writes the pdb.

**Comment by zeffii** [comment]
![image](https://cloud.githubusercontent.com/assets/619340/6170681/7d3424b0-b2d6-11e4-8fea-887667735dae.png)

**Comment by MonZop** [comment]
Yes, if the pdb contains only one conformation (of as many chains as you want), there is no need to introduce the MODEL (and the ENDMDL).
MODELs refer to several different conformations of the same molecule. 
Water is ignored by BioBlender (the HETATM in 03 are the water oxygens): they are there because the experiment revealed their location, and are probalby 'stable waters'. We can safely ingore them

**Comment by zeffii** [comment]
It might be necessary to preprocess the file (it's a fast operation), and classify it. I've pushed another code change, but am less than confident that it will handle all _unseen_ pdb files flawlessly.

**Comment by MonZop** [comment]
meaning that i should download the master again , beacause it is changed?

**Comment by zeffii** [comment]
I'm only pushing new code into `master_mff_experiments` now. (development happens in branches only)

I thought master was OK,  please don't have multiple BioBlender addons enabled at the same time, even if they have different names. There should be no errors whatsoever in the console / terminal when you start up blender. (If there are, then we need to sort that out first)

Then try the latest master-mff-experiments.

**Comment by zeffii** [comment]
I had never used the View panel before. So please guide me here.

Click on an atom, and then in View clicking on +mainChain should select other atoms too? (of that chain?)
I notice that after the surface is generated using View, that if one hides the surface, then clicks on an atom, then picks any other option that view list, it will unhide the surface (I imagine this is not desirable behaviour)

**Comment by MonZop** [comment]
The idea is to show the entire protein in one of 4 modes, that's why clicking on a single atom highlight all atoms of the aminoacid (the building block of a protein), but the selection affects the entire molecule.
In the rare case that Hydrogens are imported, the default view shows all atoms, including hydrogens. Options Main Chain, and Side Chain simply hyde the unwanted atoms.
The option Surface builds the mesh around all atoms, leaving them visible inside , as one can see by moving the surface away or by hyding it.
Once the surface is built, selecting any other option, should hyde the surface and reveal the requested atoms (either all or only some of them).
The behavior that you observe is not what should happen.
